### PR TITLE
Improve journal check so that it reports suspicious log lines directly (#22217)

### PR DESCRIPTION
**Pull request by plumps**
@flyingcircusio/release-managers

- added a black and whitelist pattern file (check_logfiles.yaml)
- part of fc.sensuplugins

### CODE REVIEW

File: nixos/modules/flyingcircus/services/sensu/client.nix [hunk 1]
```diff
@@ -246,7 +246,7 @@ in {
       };
       journal = {
         notification = "Errors in journal in the last 10 minutes";
-        command = "check-journal.rb -q '([Ee]rror|[Ee]xception)' -s -10minutes";
+        command = "${pkgs.fcsensuplugins}/bin/check_journal -v --journalctl 'journalctl -a --no-pager --since \"10 minutes ago\" ' ${pkgs.fcsensuplugins}/share/check_logfiles.yaml";
         interval = 600;
```
Comment (ckauhaus) [inline nixos/modules/flyingcircus/services/sensu/client.nix]: check_logfiles.yaml should be part of the service not the package as it is site-specific configuration.

File: nixos/modules/flyingcircus/packages/fcsensuplugins/fc/sensuplugins/journal.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+        # create a list of match candidates by applying the rules
+        warning_hits = [
+            entry
+            for rule in patterns['warningpatterns']
+            for entry in log
+            if re.search(rule, entry)
+        ]
+        # reducing them by exceptions of the rule
+        if warning_hits: warning_hits = [
```
Comment (ckauhaus) [inline nixos/modules/flyingcircus/packages/fcsensuplugins/fc/sensuplugins/journal.py]: Please format Python source according to PEP8 (and install a check plugin into your editor!)

**Comment by ckauhaus** [comment]
The code doesn't do as expected when several exception rules are present. Going to discuss the bug and potential fixes directly with MB.

**Comment by plumps** [comment]
@ckauhaus  Please re-view https://github.com/flyingcircusio/nixpkgs/pull/51/commits/8aa45827a1048c60b05c42aae63b6e3bd866682d

**Pull request merged**